src/types.js

```diff
@@ -156,7 +156,12 @@ export type Identity = $Diff<Auth, {| apiKey: string |}>;
 export type EmojiType = 'image' | 'unicode';
 
 /** An emoji, in a shape we can pass to @zulip/shared */
-export type EmojiForShared = {| emoji_type: EmojiType, emoji_name: string, emoji_code: string |};
+export type EmojiForShared = {|
+  reaction_type: ReactionType,
```

Comment:
This change, and the other changes it drives, should be possible to put in a commit that comes before the `@zulip/shared` upgrade, so that the latter is purely a dependency upgrade.

That's reflected in the inexact object type in that library's type definitions:
https://github.com/zulip/zulip-mobile/pull/5508#discussion_r1013489877